linera-chain/src/manager/multi.rs

```diff
@@ -1,6 +1,55 @@
+//! confirmed, a new chain manager is created for the next block height. It manages the consensus
+//! state until a new block is confirmed. As long as less than a third of the validators are faulty,
+//! it guarantees that at most one `ConfirmedBlock` certificate will be created for this height.
+//! There are two modes of operation:
+//!
+//! * In cooperative mode, all chain owners can propose blocks at any time. The protocol is
+//!   guaranteed to eventually confirm a block as long as no chain owner continuously actively
+//!   prevents progress.
+//! * In leader rotation mode, chain owners take turns at proposing blocks. It can make progress
+//!   as long as at least one owner is honest, even if other owners try to prevent it.
+//!
+//! ## Safety, i.e. at most one block will be confirmed
+//!
+//! In both modes is guaranteed as follows:
+//!
+//! * Validators (honest ones) never cast a vote if they have already cast any vote in a later
+//!   round.
+//! * Validators never vote for a `ValidatedBlock` **A** in round **r** if they have voted for a
+//!   _different_ `ConfirmedBlock` **B** in an earlier round **s** ≤ **r**, unless there is a
+//!   `ValidatedBlock` certificate (with a quorum of validator signatures) for **A** in some round
+//!   between **s** and **r**.
+//! * Validators only vote for a `ConfirmedBlock` if there is a `ValidatedBlock` certificate for the
+//!   same block in the same round.
+//!
+//! This guarantees that once a quroum votes for some `ConfirmedBlock`, there can never be a
+//! `ValidatedBlock` certificate (and thuse also no `ConfirmedBlock` certificate) for a different
+//! block in a later round. So if there are two different `ConfirmedBlock` certificates, they may
+//! be from different rounds, but they are guaranteed to contain the same block.
+//!
+//!
+//! ## Liveness, i.e. some block will eventually be confirmed
+//!
+//! In cooperative mode, if there is contention, the owners need to agree on a single owner as the
+//! next proposer. That owner should then download all highest-round certificates and block
+//! proposals known to the honest validators. They can then make a proposal in a round higher than
+//! all previous proposals. If there is any `ValidatedBlock` certificate they must include the
+//! highest one in their proposal, and propose that block. Otherwise they can propose a new block.
+//! Now all honest validators are allowed to vote for that proposal, and eventually confirm it.
+//!
+//! In leader-based mode, an honest owner should subscribe to notifications from all validators,
+//! and follow the chain. Whenever another leader's round takes too long, they should request
+//! timeout votes from the validators to make the next round begin. Once the honest owner becomes
+//! the round leader, they should update all validators, so that they all agree on the current
+//! round. Then they download the highest `ValidatedBlock` certificate known to any honest validator
```

Comment:
Curious if we could remove this step (and just wait instead?) given that we're assuming a subscription to all validators already. Also, it's not needed for the round `0`.